From 6432b34887c8c15f6d854983ffa95699654e6f3e Mon Sep 17 00:00:00 2001
From: Andrew Wyper <andrewx.wyper@intel.com>
Date: Tue, 16 Feb 2016 15:17:21 +0100
Subject: [PATCH 1/2] KW fix for NFC Driver

Null check was included for the return returned from call to function
'irq_to_desc'

Change-Id: I23192b4367fe18eec4f12ae78fef2ecd5b332a1c
Signed-off-by: Andrew Wyper <andrewx.wyper@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18670
Reviewed-on: https://android.intel.com:443/470979
---
 drivers/staging/pn544/pn544.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/staging/pn544/pn544.c b/drivers/staging/pn544/pn544.c
index 5aa5cc3..f848589 100644
--- a/drivers/staging/pn544/pn544.c
+++ b/drivers/staging/pn544/pn544.c
@@ -222,6 +222,9 @@ static unsigned int pn544_dev_poll(struct file *file, poll_table *wait)
 	struct device *i2c_dev = &pn544_dev->client->dev;
 	struct irq_desc *desc = irq_to_desc((pn544_dev->client->irq));
 
+	if (desc == NULL)
+		return POLLERR;
+
 	if (!gpiod_get_value(pn544_dev->irq_gpio)) {
 		dev_dbg(i2c_dev, "%s : Waiting on available input data.\n",
 				__func__);
-- 
1.9.1

